cognite/client/utils/_identifier.py

```diff
@@ -54,11 +62,21 @@ def as_dict(self, camel_case: bool = True) -> Dict[str, T_ID]:
     def as_tuple(self, camel_case: bool = True) -> Tuple[str, T_ID]:
         return self.name(camel_case), self.__value
 
-    def __str__(self) -> str:
-        return repr(self)
 
-    def __repr__(self) -> str:
-        return f"{type(self).__name__}({self.__value!r})"
+class DataModelIdentifier(IdentifierCore):
+    def __init__(self, space: str, external_id: str | None = None):
+        self.space = space
+        self.external_id = external_id
+
+    def as_dict(self, camel_case: bool = True) -> dict[str, str]:
+        output = {"space": self.space}
+        if self.external_id is None:
+            return output
+        key = "externalId" if camel_case else "external_id"
+        return {**output, key: self.external_id}
+
+    def as_primitive(self) -> None:  # type: ignore [override]
+        raise AttributeError(f"Not supported for {type(self).__name__} implementation")
```

Comment:
It would make more sense to remove this from the abstract class and only implement it on `Identifier`, or does that not work for some reason?